stdlib/asm/crypto/miden_vm_verifier/deep_queries.masm

```diff
@@ -0,0 +1,276 @@
+    exec.constants::aux_trace_com_ptr mem_loadw
+
+    ## 2) Get the leaf against the auxiliary trace commitment for the current query
+    dup.9
+    dup.9
+    mtree_get
+    exec.constants::tmp3 mem_storew
+    adv.push_mapval
+    #=> [L, R, ptr_x, ptr_alpha_inv, acc1, acc0, depth, index, query_ptr, ...]
+
+    ## 3) Load the values of the auxiliary segment of the execution trace at the current query.
+    
+    ### a) Set up the stack
+    exec.constants::zero_word_ptr mem_loadw
+    swapw
+    padw
+    #=> [Y, Y, C, ptr_x, ptr_alpha_inv, acc1, acc0, depth, index, query_ptr, ...]
+
+    ### b) Load the first 4 columns as a batch of 4 quadratic extension field elements.
+    exec.load_aux_segment_execution_trace
+    #=> [Y, D, C, ptr_x, ptr_alpha_inv, acc1, acc0, depth, index, query_ptr, ...]
+
+    ## 4) Load the leaf value we got using `mtree_get` and compare it against the hash
+    ##    we just computed
+    exec.constants::tmp3 mem_loadw
+    assert_eqw
+    #=> [Y, ptr_x, ptr_alpha_inv, acc1, acc0, depth, index, query_ptr, ...]
+end
+
+#! Computes the DEEP composition polynomial FRI queries.
+#!
+#! Input: [...]
+#! Output: [...]
+export.compute_deep_composition_polynomial_queries
+
+    exec.constants::get_fri_queries_address
+    # => [query_ptr, ...]
+
+    padw exec.constants::ood_fixed_term_horner_evaluations_ptr mem_loadw
+    # => [q_gz_1, q_gz_0, q_z_1, q_z_0, query_ptr, ...]
+    # => [W, query_ptr, ...] where W is [q_gz_1, q_gz_0, q_z_1, q_z_0]
+
+    # Get pointer to help test for the last query to be processed
+    exec.constants::fri_com_ptr
+    # => [query_end_ptr, W, query_ptr, ...]
+    dup.5
+    # => [query_ptr, query_end_ptr, W, query_ptr, ...]
+
+    # Store pointers to alpha, memory region to store trace row for current query and a fresh
+    # accumulator value i.e., (0, 0).
+    push.0.0
+    exec.constants::deep_rand_alpha_nd_ptr
+    exec.constants::current_trace_row_ptr
+    exec.constants::tmp2 mem_storew
+
+    # Compute the negations of z and gz where z is the OOD point
+    # We do it here as this computation is common to all queries.
+    exec.constants::z_ptr mem_loadw
+    # => [zN_1, zN_0, z1, z0, query_ptr, query_end_ptr, W, query_ptr, ...]
+    drop drop
+    neg swap neg  
+    # => [-z0, -z1, query_ptr, query_end_ptr, W, query_ptr, ...]
+    dup.1 exec.constants::get_trace_domain_generator mul
+    # => [-gz1, -z0, -z1, query_ptr, query_end_ptr, W, query_ptr, ...]
+    swap
+    # => [-z0, -gz1, -z1, query_ptr, query_end_ptr, W, query_ptr, ...]
+    dup exec.constants::get_trace_domain_generator mul
+    # => [-gz0, -z0, -gz1, -z1, query_ptr, query_end_ptr, W, query_ptr, ...]
+    swap
+    # => [-z0, -gz0, -gz1, -z1, query_ptr, query_end_ptr, W, query_ptr, ...]
+    # Save to temporary location `tmp1` for later use when computing the denominators
+    exec.constants::tmp1 mem_storew
+    # => [Y, query_ptr, query_end_ptr, W, query_ptr, ...]                   # (Cycles: 28)
```

Comment:
It seems like a bit part of this code is just loading values onto the stack and it doesn't seem like it is "computation-specific". If so, maybe we could pre-load the stack to before calling this procedure? Basically, maybe the signature could be something like:
```
#! Inputs:  []
#! Output: [-z0, -gz0, -gz1, -z1, query_ptr, query_end_ptr, W, query_ptr]
```

Reply:
Good point! I have factored out that code into a procedure and moved it to the generic verifier.